### ISSUE TITLE
INP001 outputs the code of the file which is part of an implicit namespace package

### ISSUE THREAD

**Issue by adamtheturtle** [astral-sh/ruff#12291]
ruff 0.4.10

This makes me think that the code has an error in it, that I am looking for, but that is not the case.

**Comment by zanieb** [comment]
Hi! Can you share a brief example?

**Comment by adamtheturtle** [comment]
Sure @zanieb ,

I create:

`foo/bar/example.py`

with the contents:

```python
STRING = "This code is not relevant"
STRING1 = "This code is not relevant"
STRING2 = "This code is not relevant"
STRING3 = "This code is not relevant"
```

```
> ruff check foo/bar/
foo/bar/example.py:1:1: INP001 File `foo/bar/example.py` is part of an implicit namespace package. Add an `__init__.py`.
  |
1 | STRING = "This code is not relevant"
  |  INP001
2 | STRING1 = "This code is not relevant"
3 | STRING2 = "This code is not relevant"
  |
```

I infer from the output that there is a problem to be fixed on line 1 of the file.

**Comment by dhruvmanila** [comment]
I think file level diagnostics by default uses an empty range at the start of the file which is what you're seeing here.

**Comment by charliermarsh** [comment]
Lets omit the code frame for empty files?

**Comment by dhruvmanila** [comment]
> Lets omit the code frame for empty files?

Yeah, that seems reasonable. I think we would want to check for an empty range at byte 0 (`TextRange::new(0, 0)`) which is where this is being raised.
